specification/columns/capacityreservationid.md

```diff
@@ -0,0 +1,37 @@
+# Capacity Reservation ID
+
+A Capacity Reservation ID is the identifier assigned to a [*capacity reservation*](#glossary:capacity-reservation) by the provider. Capacity Reservation ID is commonly used for scenarios to allocate charges for capacity reservation usage.
+
+The CapacityReservationId column adheres to the following requirements:
+
+* CapacityReservationId MUST be present in the billing data when the provider supports *capacity reservations* and MUST be of type String.
+* CapacityReservationId SHOULD NOT be null when a charge is related to a capacity reservation.
+* CapacityReservationId MUST NOT be null when a charge represents the unused portion of a *capacity reservation*.
+* CapacityReservationId MUST be null when a charge is not related to a *capacity reservation*.
+* CapacityReservationId SHOULD be a fully-qualified identifier that ensures global uniqueness within the provider.
```

Comment:
```suggestion
* CapacityReservationId MUST ensure global uniqueness within the provider and SHOULD be a fully-qualified identifier.
```

Reply:
That would be great, but unfortunately, we have not enforced uniqueness for either ResourceId or CommitmentDiscountId. I assume we still allow non-unique values for a reason.

A quote from ResourceId for reference:

> ResourceId SHOULD be a fully-qualified identifier that ensures global uniqueness within the provider

Reply:
CommitmentDiscountId does require uniqueness: `CommitmentDiscountId MUST be unique within the provider.` I would like to see us doing the same here. I like Chris' "fully qualified" amendment as well.

Reply:
Yeah Im good with unique within the provider.

Reply:
@flanakin @cnharris10 @AWS-ZachErdman

REMINDER: We've changed the CommitmentDiscountId requirement in the CommitmentUtilization PR #400 to align it with the ResourceId requirement.

v 1.0: 

> CommitmentDiscountId MUST be unique within the provider.

PR #400:

> CommitmentDiscountId SHOULD be a fully-qualified identifier that ensures global uniqueness within the provider.

I'm fully supportive of tightening the requirements for both CommitmentDiscountId and CapacityReservationId, and I hope the ResourceId will also become unique within the provider. :)